frontend/packages/operator-lifecycle-manager/src/components/descriptors/index.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { DetailsItem } from '@console/internal/components/utils';
+import { getSchemaAtPath } from '@console/shared';
+import { Descriptor, DescriptorType } from './types';
+import { K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { JSONSchema6 } from 'json-schema';
+import { SpecDescriptorDetailsItem } from './spec';
+import { StatusDescriptorDetailsItem } from './status';
+import { withFallback } from '@console/shared/src/components/error/error-boundary';
+
+export const DescriptorDetailsItem = withFallback<DescriptorDetailsItemProps>(
+  ({ descriptor, model, obj, onError, schema, type }) => {
+    const propertySchema = getSchemaAtPath(schema, descriptor.path);
+    const description = descriptor?.description || propertySchema?.description;
+    const fullPath = [type, ..._.toPath(descriptor.path)];
+    const label = descriptor.displayName || propertySchema?.title || _.startCase(_.last(fullPath));
+    const value = _.get(obj, fullPath, descriptor.value);
+    switch (type) {
+      case DescriptorType.spec:
+        return (
+          <SpecDescriptorDetailsItem
+            description={description}
+            descriptor={descriptor}
+            label={label}
+            model={model}
+            obj={obj}
+            onError={onError}
+            fullPath={fullPath}
+            value={value}
+          />
+        );
+      case DescriptorType.status:
+        return (
+          <StatusDescriptorDetailsItem
+            description={description}
+            descriptor={descriptor}
+            label={label}
+            model={model}
+            obj={obj}
+            onError={onError}
+            fullPath={fullPath}
+            value={value}
+          />
+        );
```

Comment:
Assuming the props are the same you could avoid some duplication with the following. TypeScript should still check the props.

```suggestion
    const descriptorProps = { description, descriptor, label, model, obj, onError, fullPath, value };
    switch (type) {
      case DescriptorType.spec:
        return <SpecDescriptorDetailsItem {...descriptorProps} />;
      case DescriptorType.status:
        return <StatusDescriptorDetailsItem {...descriptorProps} />;
```